database-commons/src/main/java/io/cdap/plugin/db/source/DataDrivenETLDBInputFormat.java

```diff
@@ -64,6 +69,49 @@ public static void setInput(Configuration conf,
     new ConnectionConfigAccessor(conf).setAutoCommitEnabled(enableAutoCommit);
   }
 
+  /**
+   * A wrapper around the superclass getSplits method. This is required for testing,
+   * as mocking a method that calls its superclass implementation is challenging.
+   * This wrapper allows unit tests to override this method and mock the behavior of
+   * the superclass call, isolating the logic within the overridden getSplits method.
+   *
+   * @param job The job context.
+   * @return The list of input splits generated by the parent class.
+   * @throws IOException If an error occurs while getting the splits.
+   */
+  @VisibleForTesting
+  List<InputSplit> getBaseSplits(JobContext job) throws IOException {
+    return super.getSplits(job);
+  }
+
+  @Override
+  public List<InputSplit> getSplits(JobContext job) throws IOException {
+    List<InputSplit> splits = getBaseSplits(job);
+
+    // Check if a split for NULL values or an all-encompassing split ("1=1") already exists.
+    boolean shouldAddNullSplit = splits.stream()
+        .map(DataDrivenDBInputFormat.DataDrivenDBInputSplit.class::cast)
+        .map(DataDrivenDBInputFormat.DataDrivenDBInputSplit::getLowerClause)
+        .noneMatch(lowerClause ->
+            !Strings.isNullOrEmpty(lowerClause)
+                && (lowerClause.contains("IS NULL") || lowerClause.contains("1=1")));
+
+    if (shouldAddNullSplit) {
+      // Create a dedicated split to handle potential NULL values in the split-by column.
+      String splitByColumn = getDBConf().getInputOrderBy();
+      String clause = String.format("%s IS NULL", splitByColumn);
```

Comment:
Please add quotes around the column name to ensure casing is preserved in query.